launcher/minecraft/mod/ResourceFolderModel.cpp

```diff
@@ -158,11 +166,55 @@ bool ResourceFolderModel::installResource(QString original_path)
     return false;
 }
 
-bool ResourceFolderModel::uninstallResource(QString file_name)
+bool ResourceFolderModel::installResource(QString path, ModPlatform::IndexedVersion& vers)
+{
+    if (vers.addonId.isValid()) {
+        ModPlatform::IndexedPack pack{
+            vers.addonId,
+            ModPlatform::ResourceProvider::FLAME,
+        };
+
+        QEventLoop loop;
```

Comment:
I know we use this pattern elsewhere but I think it would be better to not to make sub loops. it's caused problems of late.
IMO it's better to have a `postInstallResource` slot and use a callback function for the return value of the operation.